src/planscape/planning/views.py

```diff
@@ -366,36 +370,100 @@ def get_scenario_download_by_id(request: HttpRequest) -> HttpResponse:
 
     TODO: maybe generate a unique key and store that for each output dir name when we create it?
     """
-    try:
-        # Ensure that the user is logged in.
-        user = _get_user(request)
-        if user is None:
-            return HttpResponseBadRequest(
-                "Unauthorized. User is not logged in.", status=401
-            )
+    # Ensure that the user is logged in.
+    user = _get_user(request)
+    if user is None:
+        return HttpResponse(
+            "Unauthorized. User is not logged in.",
+            status=401,
+        )
 
-        scenario = Scenario.objects.select_related("planning_area__user").get(
-            id=request.GET["id"]
+    scenario = (
+        Scenario.objects.select_related("planning_area__user")
+        .filter(id=request.GET["id"])
+        .first()
+    )
+
+    if not scenario:
+        return HttpResponse(
+            "Scenario matching query does not exist.",
+            status=404,
+        )
+
+    # Ensure that current user is associated with this scenario
+    if scenario.planning_area.user.pk != user.pk:
+        return HttpResponse(
+            "Scenario matching query does not exist.",
+            status=404,
```

Comment:
I was also wondering if we should send a 404 here or maybe a 403 -- a 403 is more accurate, and it would help us with debugging, but a 404 would hide the fact that this resource actually exists, if we want to obfuscate this for some reason.

Reply:
Yeah, I agree, lets just leave it as a 404.